### ISSUE TITLE
`mypy` crash due to `CachingProtocolMeta`

### ISSUE THREAD

**Issue by lundberg** [antonagestam/phantom-types#243]
I'm suddenly getting this on cached `mypy` runs ...

```console
Traceback (most recent call last):
  File "/app/venv/bin/mypy", line 8, in <module>
    sys.exit(console_entry())
  File "/app/venv/lib/python3.10/site-packages/mypy/__main__.py", line 15, in console_entry
    main()
  File "/app/venv/lib/python3.10/site-packages/mypy/main.py", line 95, in main
    res, messages, blockers = run_build(sources, options, fscache, t0, stdout, stderr)
  File "/app/venv/lib/python3.10/site-packages/mypy/main.py", line 174, in run_build
    res = build.build(sources, options, None, flush_errors, fscache, stdout, stderr)
  File "/app/venv/lib/python3.10/site-packages/mypy/build.py", line 193, in build
    result = _build(
  File "/app/venv/lib/python3.10/site-packages/mypy/build.py", line 276, in _build
    graph = dispatch(sources, manager, stdout)
  File "/app/venv/lib/python3.10/site-packages/mypy/build.py", line 2903, in dispatch
    process_graph(graph, manager)
  File "/app/venv/lib/python3.10/site-packages/mypy/build.py", line 3280, in process_graph
    process_fresh_modules(graph, prev_scc, manager)
  File "/app/venv/lib/python3.10/site-packages/mypy/build.py", line 3361, in process_fresh_modules
    graph[id].fix_cross_refs()
  File "/app/venv/lib/python3.10/site-packages/mypy/build.py", line 2112, in fix_cross_refs
    fixup_module(self.tree, self.manager.modules, self.options.use_fine_grained_cache)
  File "/app/venv/lib/python3.10/site-packages/mypy/fixup.py", line 53, in fixup_module
    node_fixer.visit_symbol_table(tree.names, tree.fullname)
  File "/app/venv/lib/python3.10/site-packages/mypy/fixup.py", line 125, in visit_symbol_table
    self.visit_type_info(value.node)
  File "/app/venv/lib/python3.10/site-packages/mypy/fixup.py", line 87, in visit_type_info
    info.declared_metaclass.accept(self.type_fixer)
  File "/app/venv/lib/python3.10/site-packages/mypy/types.py", line 1283, in accept
    return visitor.visit_instance(self)
  File "/app/venv/lib/python3.10/site-packages/mypy/fixup.py", line 196, in visit_instance
    inst.type = lookup_fully_qualified_typeinfo(
  File "/app/venv/lib/python3.10/site-packages/mypy/fixup.py", line 337, in lookup_fully_qualified_typeinfo
    assert (
AssertionError: Should never get here in normal mode, got Var:phantom._utils.types.CachingProtocolMeta instead of TypeInfo
```

Seems to be an issue with `CachingProtocolMeta` meta class being exposed as `Any` when looking at the `mypy` cache ...

```json
{
    "CachingProtocolMeta": {
        ".class": "SymbolTableNode",
        "kind": "Gdef",
        "module_hidden": true,
        "module_public": false,
        "node": {
            ".class": "Var",
            "flags": [
                "is_suppressed_import",
                "is_ready",
                "is_inferred"
            ],
            "fullname": "phantom._utils.types.CachingProtocolMeta",
            "name": "CachingProtocolMeta",
            "type": {
                ".class": "AnyType",
                "missing_import_name": "phantom._utils.types.CachingProtocolMeta",
                "source_any": null,
                "type_of_any": 3
            }
        }
    }
}
```

**Comment by antonagestam** [comment]
@flaeppe @lundberg Thanks for reporting and providing the fix 🙏

I merged #244, but I'll keep this issue open until we verify it's mitigated. I guess it could makes sense to report it to mypy as well though, I know they're working hard to reduce crashes for 1.0 ;)

If you're able to verify by installing the merged commit directly, that'd be very much appreciated!

**Comment by antonagestam** [comment]
Ah, so it seems this is really a regression in mypy `0.99{0,1}`. I assume that's what you're running on?

https://github.com/antonagestam/phantom-types/actions/runs/3491332817/jobs/5843838849

**Comment by antonagestam** [comment]
New fix is in https://github.com/antonagestam/phantom-types/pull/249, and released in 1.1.0. Let me know if it doesn't solve the issue on your side. Thanks again!

**Comment by flaeppe** [comment]
> Ah, so it seems this is really a regression in mypy `0.99{0,1}`. I assume that's what you're running on?
> 
> https://github.com/antonagestam/phantom-types/actions/runs/3491332817/jobs/5843838849

Yep true, we were running `0.990` when it started breaking

> New fix is in #249, and released in 1.1.0. Let me know if it doesn't solve the issue on your side. Thanks again!

We'll let you know once we've bumped
